src/main/java/org/jenkinsci/plugins/workflow/steps/RetryStepExecution.java

```diff
@@ -61,6 +65,18 @@ public void onSuccess(StepContext context, Object result) {
         @Override
         public void onFailure(StepContext context, Throwable t) {
             try {
+                Run run = context.get(Run.class);
+                if (run != null && t instanceof FlowInterruptedException) {
+                    InterruptedBuildAction action = run.getAction(InterruptedBuildAction.class);
+                    if (action != null) {
+                        for (CauseOfInterruption cause : action.getCauses()) {
```

Comment:
No, use `FlowInterruptedException.getCauses()`, then you do not need to refer to `Run` or `InterruptedBuildAction` at all.